staging/src/k8s.io/apiserver/pkg/admission/configuration/mutating_webhook_manager.go

```diff
@@ -66,8 +71,22 @@ func (m *mutatingWebhookConfigurationManager) Webhooks() []webhook.WebhookAccess
 	return m.configuration.Load().([]webhook.WebhookAccessor)
 }
 
+// HasSynced returns true when the manager is synced with existing webhookconfig
+// objects at startup-- which means the informer is synced and either has no items
+// or updateConfiguration() has completed.
 func (m *mutatingWebhookConfigurationManager) HasSynced() bool {
-	return m.hasSynced()
+	if !m.hasSynced() {
+		return false
+	}
+	if m.initialUpdateConfigurationCompleted.Load().(bool) {
+		// the informer has synced and configuration has been updated
+		return true
+	}
+	if configurations, err := m.lister.List(labels.Everything()); err == nil && len(configurations) == 0 {
+		// the informer has synced and we don't have any items
+		return true
+	}
```

Comment:
in a cluster without any admission webhooks configured, this List() call will get invoked for every request forever... ideally we could do something cheaper.

if we rename `initialUpdateConfigurationCompleted` to `initialConfigurationCompleted`, we could set it to true here, since the synced lister had no items and we already started the configuration with an empty list

```suggestion
	if configurations, err := m.lister.List(labels.Everything()); err == nil && len(configurations) == 0 {
		// the empty list we initially stored is valid to use
		m.initialConfigurationCompleted.Store(true)
		// the informer has synced and we don't have any items
		return true
	}
```

then subsequent checks would be able to take the fast path on the atomic boolean

Reply:
@liggitt Good point! Done.